CONTRIBUTING.md

```diff
@@ -0,0 +1,268 @@
+commenting on either and engaging in discussions about other contributions to name a few. This
+document is intented as a set of guidelines to help your contribution get accepted faster, maintain
+a high standard, and to help us (the maintainers) set a few ground rules for working with us.
+
+If you have any questions about the contents of this document, the code, or how to contribute, please
+drop us a line on [Gitter](https://gitter.im/opentk/opentk). We'll be happy to answer as best we can.
+
+#### Table of Contents
+1. [Things to keep in mind](#things-to-keep-in-mind)
+2. [Setting Up](#setting-up)
+3. [Pull Requests](#pull-requests)
+	1. [Bug Fixes](#bug-fixes)
+	2. [New Features](#new-features)
+	3. [Cosmetic & Stylistic Changes](#cosmetic-&-stylistic-changes)
+	4. [Breaking Changes](#breaking-changes)
+4. [Bug Reports](#bug-reports)
+5. [Discussions & Suggestions](#discussions-&-suggestions)
+
+
+## Things to keep in mind
+Like a lot of other modern projects, OpenTK is written for multiple platforms and operating systems.
+Therefore, it's important to keep this in mind when contributing to the project - otherwise, it may 
+make accepting your contribution much more difficult. You'll want to consider that the bug you're 
+experiencing might not be present on other platforms or system configurations, or that your pull
+request doesn't take all platforms into account. Sometimes this important, sometimes it's not.
+
+OpenTK is also (as previously mentioned) a very large codebase which has seen a lot of people and a 
+lot of styles over the years. This is reflected in the deeper, darker parts of the codebase where
+mixed styles, weird naming, bizarre code and eldritch sorcery abound. What may seem like a small change
+on the surface could lead you down on a path of unraveling one thread after another, and what started
+off as a simple bug fix could transform into a lot of headscratching.
+
+To make this at least somewhat easier, here's a few concrete general tips which you should stick to:
+
+* Always consider cross-platform gotchas.
+* Always work in small, iterative chunks which you can easily describe.
+* Avoid cosmetic or visual changes, unless your contribution is strictly focused on that.
+* Don't be afraid to ask, especially before diving in. There might be someone else working on the very
+same thing already!
+* Consider how your contribution might affect other contributions. Sometimes one change will break another
+if you're not careful.
+
+In terms of these guidelines, the terminology is as follows:
+* Must: If your contribution does not follow this rule, it will not be accepted.
+* Should: If your contribution does not follow this rule, it has a lower chance of being accepted.
+* May: If your contribution does not follow this rule, it's probably not going to matter that much. 
+It'd be a nice touch, though.
+
+With that in mind, check the following sections for more concrete and direct guidelines.
```

Comment:
Just "check the following sections for guidelines." the elaboration about concrete and direct isn't needed.